Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -94,4 +115,77 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
         this._object.y,
       ])
     );
+//  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
+  var result = [];
+  var center = [this._object.x + this._radius, this._object.y + this._radius];
+  var halfOfDiag = Math.sqrt(2) * this._radius;
+
+  this._manager.getAllObstaclesAround(this._object, this._radius, result);
+  var hitBoxes = result.map(function (item) {
+    return item.owner.hitBoxes;
+  });
+  hitBoxes.push(this._object.getHitBoxes());
+
+  var polygons = hitBoxes.flat(1);
+  var vertices = polygons.map(function (poly) {
+    return poly.vertices;
+  }).flat(1);
```

Comment:
We'll replace `.flat(1)` in the future by (potentially) nested loops, because flat is ES6. For now, concentrate on making this work, add this to a to do list in the trello or in the code :)